### PR TITLE
Update optional-features.md to include no-implicit-route-model

### PR DESCRIPTION
**Pull request by achambers**

### CODE REVIEW

**Comment by mansona** [comment]
This should be merged when Ember 5.7 is released. I've added it to a new "milestone" and we can discuss adding version milestones to our release process 👍

File: guides/release/configuring-ember/optional-features.md [hunk 1]
```diff
@@ -33,9 +33,9 @@ Usage:
 
 Available features:
 
-  some-example-feature (Default: true)
-    A description of the feature goes here
-    More information: <link to an RFC>
+  no-implicit-route-model (Default: false)
+    Removes the default record loading behaviour on Ember's Route.
```
Comment (mansona) [inline guides/release/configuring-ember/optional-features.md:37]:
```suggestion
    Removes the default record loading behavior on Ember's Route.
```

This suggestion makes me sad 🙃 🇮🇪

File: guides/release/configuring-ember/optional-features.md [hunk 2]
```diff
@@ -33,9 +33,9 @@ Usage:
 
 Available features:
 
-  some-example-feature (Default: true)
-    A description of the feature goes here
-    More information: <link to an RFC>
+  no-implicit-route-model (Default: false)
```
Comment (achambers) [inline guides/release/configuring-ember/optional-features.md:36]:
```suggestion
  no-implicit-route-model (Default: true)
```

File: guides/release/configuring-ember/optional-features.md [hunk 3]
```diff
@@ -46,11 +46,23 @@ Let us disable an optional feature to see what happens. Substitute `some-example
...
+$ ember feature:disable no-implicit-route-model
+Disabled no-implicit-route-model. Be sure to commit config/optional-features.json to source control!
 ```
 
 As we can see from the warning, `@ember/optional-features` has created a file in `config/optional-features.json` to store the configuration for your project.
 We commit it to our repository and we are off to the races!
 
+### no-implicit-route-model
+
+This feature is related to esoteric features of route model loading that you likely do not use, or know exist, and [have been deprecated](https://deprecations.emberjs.com/v5.x#toc_deprecate-implicit-route-model) in `5.3.0`. They are due to be removed in `6.0.0`. To clear the deprecation, you can enable this feature.
```
Comment (mansona) [inline guides/release/configuring-ember/optional-features.md:58]:
```suggestion
This feature is related to esoteric features of route model loading that you likely do not use, or know exist, and [have been deprecated](https://deprecations.emberjs.com/id/deprecate-implicit-route-model/) in `5.3.0`. They are due to be removed in `6.0.0`. To clear the deprecation, you can enable this feature.
```

File: guides/release/configuring-ember/optional-features.md [hunk 4]
```diff
@@ -46,11 +46,23 @@ Let us disable an optional feature to see what happens. Substitute `some-example
...
+
+With this feature disabled, Ember will automatically load a route's model if the `model` hook has not been implemented. In this case, Ember will attempt to try a few things before rendering this route's template.
+
+1. If there is a `store` property on your route, it will attempt to call its `find` method. Assuming you have ember-data installed, you may be expecting this. The arguments will be extracted from the params. For example, if a dynamic segment is `:post_id`, there exists logic to split on the underscore and find a record of type post.
+
+2. As a fallback, it will attempt to define a `find` method and use your Model instance's `find` method to fetch. If a Model cannot be found or if the found Model does not have a find method, an assertion is thrown.
+
+Enabling this optional feature will remove this implicit model loading behavior and leave it to you to implement if and when you need it.
+
 <!-- eof - needed for pages that end in a code block  -->
```
Comment (mansona) [inline guides/release/configuring-ember/optional-features.md:68]:
```suggestion

```

**Review by mansona — APPROVED** [review]

**Pull request merged**